src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Patterns.cs

```diff
@@ -133,12 +133,25 @@ private bool IsIrrefutablePatternTest(BoundExpression test)
 
         private BoundExpression CompareWithConstant(BoundExpression input, BoundExpression boundConstant)
         {
-            return _factory.StaticCall(
-                _factory.SpecialType(SpecialType.System_Object),
-                "Equals",
-                _factory.Convert(_factory.SpecialType(SpecialType.System_Object), boundConstant),
-                _factory.Convert(_factory.SpecialType(SpecialType.System_Object), input)
-                );
+            var systemObject = _factory.SpecialType(SpecialType.System_Object);
```

Comment:
Meta feedback: it's really unclear where in the compiler we can, or can't, depend on special types being non-null in the compiler.  Depending on it in the local rewriter seems sane, but I've been surprised in the past.  Can we take a follow up item to get good guidance on this point?